### PR TITLE
remove useless cleanup of /atom/var/reagents

### PR DESCRIPTION
**Pull request by Kegdo**
## About The Pull Request

/datum/reagents/Destroy() does everything /datum/reagents/delete() does and more, and delete() is called only from /obj/effect/Destroy() which ITSELF is redundant with its parent /atom/movable/Destroy()

## Why It's Good For The Game

optimisation.

## Changelog
:cl:
del: removed abundant cleanup of atom/var/reagents.
/:cl:

### CODE REVIEW

**Comment by hyperioo** [comment]
@Kegdo Is the
```
for(var/datum/reagent/R in reagent_list)
	R.holder = null
```
done somewhere else?

**Comment by Kegdo** [comment]
@hyperioo no.

**Comment by hyperioo** [comment]
You could add `holder = null` to `/datum/reagents/Destroy()`

**Comment by Kegdo** [comment]
sure, here.

**Review by hyperioo — APPROVED** [review]
Hope I'm not missing something

File: code/game/objects/effects/effect_system.dm [hunk 1]
```diff
@@ -32,13 +32,6 @@ would spawn and follow the beaker, even if it is carried or thrown.
 		pixel_x += rand(-random_offset,random_offset)
 		pixel_y += rand(-random_offset,random_offset)
 
-
-
-/obj/effect/Destroy()
```
Comment (drexample) [inline code/game/objects/effects/effect_system.dm:37]: Why did you remove this?

**Review by drexample — CHANGES_REQUESTED** [review]

File: code/game/objects/effects/effect_system.dm (hunk 1 above)
Comment (Kegdo) [reply code/game/objects/effects/effect_system.dm:37]: image

**Comment by drexample** [comment]
Merging this, if chem puddles break, we're gonna shoot you into space.
